reference/helpers.go

```diff
@@ -0,0 +1,34 @@
+package reference
+
+// IsNameOnly returns true if reference only contains a repo name.
+func IsNameOnly(ref Named) bool {
+	if _, ok := ref.(NamedTagged); ok {
+		return false
+	}
+	if _, ok := ref.(Canonical); ok {
+		return false
+	}
+	return true
+}
+
+// SplitName is a helper function which parses the
+// input as a name and returns name plus tag or digest
+func SplitName(s string) (string, string, error) {
```

Comment:
I agree, I got it from here https://github.com/docker/engine-api/blob/master/types/reference/image_reference.go#L10 but even its use looks trivial. It would be better to be more explicit about parsing into a Named. It is possible for `engine-api` to use the normalized versions now as well.